app/core/util.py

```diff
@@ -73,3 +79,102 @@ def tree_table_to_json(
+
+    return physical_document_updated(db, row) + family_updated(db, row)
+
+
+def family_updated(db: Session, row: DfcRow) -> List[UpdateResult]:
+    """Identify any updates to the family relating to the row by comparing against the relevant tables in
+    the database.
+
+    Args:
+        row (DfcRow): the row from the CSV.
+        db (Session): the database session.
+
+    Returns:
+        dict[field_name: UpdateResult(field=value csv_value=value, db_value=value, update=bool, type=table_name)]
+    """
+    family_document = (
+        db.query(FamilyDocument)
+        .filter(FamilyDocument.import_id == row.cpr_document_id)
+        .scalar()
+    )
+
+    family = db.query(Family).filter(Family.id == family_document.family_id).scalar()
+
+    return [
+        result
+        for result in [
+            UpdateResult(
+                db_value=family.title,
+                csv_value=row.family_name,
+                updated=family.title != row.family_name,
+                type="Family",
+                field="title",
+            ),
+            UpdateResult(
+                db_value=family.description,
+                csv_value=row.family_summary,
+                updated=family.description != row.family_summary,
+                type="Family",
+                field="description",
+            ),
+        ]
+        if result.updated
+    ]
+
+
+def physical_document_updated(db: Session, row: DfcRow) -> List[UpdateResult]:
+    """Identify any updates to the physical document by comparing against the relevant tables in the database
+
+    Args:
+        row (DfcRow): the row from the CSV.
+        db (Session): the database session.
+
+    Returns:
+        List[UpdateResult(csv_value=value, db_value=value, update=bool, type=table_name, field=value)]"""
+    family_document = (
+        db.query(FamilyDocument)
+        .filter(FamilyDocument.import_id == row.cpr_document_id)
+        .scalar()
+    )
+
+    physical_document = (
+        db.query(PhysicalDocument)
+        .filter(PhysicalDocument.id == family_document.physical_document_id)
+        .scalar()
+    )
+
+    db_source_url = (
+        ""
+        if physical_document.source_url == '{""}'
+        else physical_document.source_url.replace("{", "")
+        .replace("}", "")
+        .split(",")[0]
+    )
+    csv_source_url = row.documents[0]
+
+    return [
+        result
+        for result in [
+            UpdateResult(
+                db_value=physical_document.title,
+                csv_value=row.document_title,
+                updated=physical_document.title != row.document_title,
+                type="PhysicalDocument",
+                field="title",
+            ),
+            UpdateResult(
+                db_value=db_source_url,
+                csv_value=csv_source_url,
+                updated=db_source_url != csv_source_url,
+                type="PhysicalDocument",
```

Comment:
Need to find a better way to define type and field, might be able to just put the type.

Reply:
Then we just query directly.

Reply:
type=Family
db.update(Family, **{field: value})